autosklearn/util/backend.py

```diff
@@ -37,13 +39,22 @@ def get_randomized_directory_names(
 ):
     random_number = random.randint(0, 10000)
     pid = os.getpid()
+    m5_hash = str(md5(
+        str(time.localtime()).encode('utf-8')
+    ).hexdigest())
+    hostname = socket.gethostname()
 
     temporary_directory = (
         temporary_directory
         if temporary_directory
         else os.path.join(
             tempfile.gettempdir(),
-            'autosklearn_tmp_%d_%d' % (pid, random_number),
+            "autosklearn_tmp_{}_{}_{}_{}".format(
```

Comment:
Looking at this code now, I'm wondering whether we should use [UUID1](https://docs.python.org/3/library/uuid.html#uuid.uuid1) with the process ID being the sequence number. Seems to be exactly what we're trying to solve here (and the random number should no longer be necessary since we take the time into account).